### DISCUSSION TITLE
struct.error: unpack requires a buffer of 11 bytes

### DISCUSSION THREAD

**Discussion by shrijayan** [bitcraze/discussions#137] [Q&A]
While we are trying to execute AI Deck Example Program named opencv-viewer.py we have completed the first step of Socket connection but there was an error resulting in us "struct.error: unpack requires a buffer of 11 bytes" 

what should we change in the python code ?

We have attached the screenshot for reference.

![Screenshot from 2022-07-11 17-02-02](https://user-images.githubusercontent.com/81805145/178255958-b03307fe-88d3-4188-829e-81c2434aa047.png)

**Comment by knmcguire** [comment]
Hi!

It seems like it is not sending the image in the right way. Have you gone through the [getting started tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) for the AIdeck?

**Reply by shrijayan** [reply]
Respected @knmcguire,
Yes, we have gone through [getting started tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) for the AI deck. 

We have done :
## Preparing the AI deck and Crazyflie firmware
**Update Crazyflie and AIdeck firmware**
1. We can find only the cfclient version only of 2020.09.1 but we can't find the version 2022.05+
2. Also we not see the output in the console are "ESP32: I (910) SYS: Initialized"

**Gap8 bootloader (for older AIdecks revisions A-C)**
After the execution of before commands if we try to run these command

docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'

it gives an error 

*DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
TARGET create
GAP8 INIT TARGET
Error: no device found
Error: unable to open ftdi device with vid 15ba, pid 002a, description 'Olimex OpenOCD JTAG ARM-USB-TINY-H', serial '*' at bus location '*'

Traceback (most recent call last):
  File "/gap_sdk/utils/gapy/gapy", line 162, in <module>
    main()
  File "/gap_sdk/utils/gapy/gapy", line 152, in main
    operationFunc(args, config, system)
  File "/gap_sdk/utils/gapy/run.py", line 184, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/gap_sdk/rtos/freeRTOS/vendors/gwt/rules/freeRTOS_rules.mk:404: flash] Error 1*

we don't know how to connect and where to connect the JTAG and JLINK ?

**Flash Wifi Example**
cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w radio://0/80/2M/E7E7E7E7E7

it is executing but still it stops here

*Reset to bootloader mode ...
Could not save cache, no writable directory
Could not save cache, no writable directory
Skipping bcAI:esp, not in the target list
Deck bcAI:gap8, reset to bootloader
| 0% Writing to bcAI:gap8 deck memory*

in this step, we were waiting more than an hour still no positive result.

![img](https://user-images.githubusercontent.com/81805145/179262693-8b59dee5-85e3-4b8e-b060-841309cbfa1e.jpeg)

Yours sincerely,
Shrijayan
https://cpluz.com/

**Reply by knmcguire** [reply]
**Update Crazyflie and AIdeck firmware** The fact that you are not able to update to the new cfclient Is probably due to the version of your python. You need to have at least [python 3.7 ](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/). Only once you managed to flash the ESP of the aideck with the firmware update, then you should continue.

**Gap8 bootloader (for older AIdecks revisions A-C)** This is only necessary if you have an old revision of the AI deck so please check the bottom if it is REV D (for which you do not need a programmer) or REV A-C (for which you do need a programmer). 

**Flash Wifi Example** This part is not possible to do without the steps before being successful, so please wait until you managed to go through the first step of 'Update Crazyflie and AIdeck firmware'
